eng/pipelines/templates/jobs/sdk-diff-tests.yml

```
@@ -0,0 +1,136 @@
+      buildId: $(DotnetDotnetBuildId)
+      artifact: ${{ parameters.buildName }}_${{ parameters.architecture }}_Artifacts
+      patterns: '**/dotnet-sdk-+([0-9]).+([0-9]).+([0-9])?(-@(alpha|preview|rc|rtm)*)-${{ parameters.targetRid }}.tar.gz' 
+      allowPartiallySucceededBuilds: true
+      allowFailedBuilds: true
+      downloadPath: $(Pipeline.Workspace)/Artifacts
+      checkDownloadedFiles: true
+    
+  - script: |
+      set -x
+
+      msft_sdk_tarball_name=$(find "$(Pipeline.Workspace)/Artifacts" -name "dotnet-sdk-*-linux-${{ parameters.architecture }}.tar.gz" -exec basename {} \;)
+
+      if [[ -z $msft_sdk_tarball_name ]]; then
+        fail "Microsoft SDK tarball does not exist in '$(Pipeline.Workspace)/Artifacts'"
+        exit 1
+      fi
+
+      sdk_tarball_name=$(find "$(Pipeline.Workspace)/Artifacts" -name "dotnet-sdk-*-${{ parameters.targetRid }}.tar.gz" -exec basename {} \;)
+
+      if [[ -z $sdk_tarball_name ]]; then
+        fail "Source-build SDK tarball does not exist in '$(Pipeline.Workspace)/Artifacts'"
+        exit 1
+      fi
+
+      eng/common/build.sh --projects $(Build.SourcesDirectory)/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/Microsoft.DotNet.SourceBuild.SmokeTests.csproj --restore
+
+      echo "##vso[task.setvariable variable=MsftSdkTarballName]$msft_sdk_tarball_name"
+      echo "##vso[task.setvariable variable=SdkTarballName]$sdk_tarball_name"
+    displayName: Prepare Tests
+    workingDirectory: $(Build.SourcesDirectory)
+
+  - script: >
+      .dotnet/dotnet 
+      test
+      $(Build.SourcesDirectory)/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/Microsoft.DotNet.SourceBuild.SmokeTests.csproj
+      --filter "FullyQualifiedName=Microsoft.DotNet.SourceBuild.SmokeTests.SdkContentTests.CompareMsftToSb"
+      --logger:'trx;LogFileName=$(Agent.JobName)_SDKDiffTests.trx'
+      --logger:'console;verbosity=detailed'
+      -e SMOKE_TESTS_MSFT_SDK_TARBALL_PATH=$(Pipeline.Workspace)/Artifacts/$(MsftSdkTarballName)
+      -e SMOKE_TESTS_SDK_TARBALL_PATH=$(Pipeline.Workspace)/Artifacts/$(SdkTarballName)
+      -e SMOKE_TESTS_SOURCEBUILT_ARTIFACTS_PATH= 
+      -e SMOKE_TESTS_WARN_SDK_CONTENT_DIFFS=false
+      -e SMOKE_TESTS_RUNNING_IN_CI=true
+      -e SMOKE_TESTS_TARGET_RID=${{ parameters.targetRid }}
+      -e SMOKE_TESTS_PORTABLE_RID=linux-x64
+      -e SMOKE_TESTS_CUSTOM_PACKAGES_PATH= 
+    displayName: Run Tests
+    workingDirectory: $(Build.SourcesDirectory)
+
+  - task: PublishTestResults@2
+    displayName: Publish Test Results
+    condition: succeededOrFailed()
+    continueOnError: true
+    inputs:
+      testRunner: vSTest
+      testResultsFiles: '*.trx'
+      searchFolder: $(Build.SourcesDirectory)/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/TestResults
+      mergeTestResults: true
+      publishRunAttachments: true
+      testRunTitle: SourceBuild_SdkComparisonTests_$(Agent.JobName)

```

Looking at the build, I'd just do this:
```suggestion
      testRunTitle: $(Agent.JobName)
```